src/Corax/Queries/TermProviders/TermProvider.Exists.cs

```diff
@@ -43,16 +46,18 @@ public bool Next(out TermMatch term)
 
         public bool GetNextTerm(out ReadOnlySpan<byte> term)
         {
-            while (_iterator.MoveNext(out Span<byte> termSlice, out var _))
+            while (_iterator.MoveNext(out var keyScope, out var _))
             {
-                int termSize = termSlice.Length;
-                if (termSlice.Length > 1)
+                var key = keyScope.Key.Decoded();
```

Comment:
Why do we need to decode for `Exists` ? 
That is just because we expose the term for the `TermMatch` ?

Reply:
Because for now `TermMatch` doesnt use the key abstraction.